doc-content/drools-docs/src/main/asciidoc/AuthoringAssets/test-designer-run-test-proc.adoc

```diff
@@ -1,13 +1,29 @@
 [id='test-designer-run-test-proc']
-= Running a test scenario
+= Running the test scenarios
 
 After creating a test scenario template and defining the test scenarios, you can run the tests to validate your business rules and data.
 
 .Procedure
-. At the top of the test scenarios designer, click *Run Test* image:AuthoringAssets/test-scenarios-run-test-icon.png[Run Test icon] icon.
-. The *Test Report* panel pops out from the right of the designer and displays the test result overview and the scenario status.
-. After the tests are executed, click *View Alerts* to open the *Alerts* panel at the bottom of the designer for the error messages.
-. In case the tests fails, review the error messages. Messages with a red cross indicate test failures where as those with a green check mark indicate successful tests.
+. To run defined test scenarios, do one of the following tasks:
+* To execute all the available test scenarios in your project inside multiple assets, in the upper-right corner of your project page, click *Test*.
 +
-In the designer, erroneous rows or cells are highlighted in red.
-. Make the necessary changes and run the test again till the scenario passes.
+.Run all the test scenarios from the project view
+image:AuthoringAssets/test-scenarios-run-all-tests-button.png[Run all the test scenarios from the project view]
++
+* To execute all the available test scenarios defined in a single `.scesim` file, at the top of the Test Scenario designer, click *Run Test* image:AuthoringAssets/test-scenarios-run-test-icon.png[Run Test icon] icon.
+* To run a single test scenario defined in a single `.scesim` file, right-click the row of the test scenario you want to run and select *Run scenario*.
+. The *Test Report* panel displays the overview of the tests and the scenario status.
++
+After the tests are executed, if the values entered in the test scenario table do not match, the expected values in the corresponding cells are highlighted.
++
+.Visual test results in Test Scenario
+image:AuthoringAssets/test-scenarios-visual-test-results.png[Visual test results in Test Scenario]
++
+
+. To review the the error message in the pop-up window, hover your mouse cursor over the highlighted cell.
++
+* To update the highlighted cell with the expected value, click `Apply`.
+* If you do not want to apply any changes to the highlighted cell, click `Keep`.
```

Comment:
This is true only if the runner is able to get the actual value to compare with the expected one. In case of other type of failures, the popover will just show the error message.
There are multiple reasons why the runner might be not able to retrieve the actual value:
- if the value in the cell is syntactically not correct (`f alse` instead of `false`)
- if the decision has not been evaluated at all because of a failure in one of the previous decision (this is DMN specifc)
- if the user is looking for a new instance with some specific values that should be generate during the execution (this is RULE specifc)

I'm not sure these are the only reason why this can happen and I don't think we should list all them in the documentation. I think it is enough to add a note to say that the correct value is not always available

Reply:
As we discussed in our meeting, I will be just documenting about the "how to view the error message on the UI" and not about the error message types.